x-pack/plugins/security_solution/server/search_strategy/security_solution/factory/hosts/details/helpers.ts

```diff
@@ -200,15 +202,30 @@ export const getHostEndpoint = async (
         ? await getHostMetaData(metadataRequestContext, id, undefined)
         : null;
 
+    const fleetAgentId = endpointData?.metadata.elastic.agent.id;
+    const [fleetAgentStatus, pendingActions] = !fleetAgentId
+      ? [undefined, {}]
+      : await Promise.all([
+          // Get Agent Status
+          agentService.getAgentStatusById(esClient.asCurrentUser, fleetAgentId),
+          // Get a list of pending actions (if any)
+          getPendingActionCounts(esClient.asCurrentUser, [fleetAgentId]).then((results) => {
+            return results[0].pending_actions;
+          }),
+        ]);
+
     return endpointData != null && endpointData.metadata
       ? {
           endpointPolicy: endpointData.metadata.Endpoint.policy.applied.name,
           policyStatus: endpointData.metadata.Endpoint.policy.applied.status,
           sensorVersion: endpointData.metadata.agent.version,
+          elasticAgentStatus: fleetAgentStatusToEndpointHostStatus(fleetAgentStatus!),
```

Comment:
Should `fleetAgentStatus` be cast as non null here?

Reply:
Yeah, its possible that the endpoint/agent might no longer be active/installed, so the API call would be undefined or in the case where endpoint data is not found. The `fleetAgentStatusToEndpointHostStatus()` will handle that by returning a default status